apollo/agent/agent.py

```diff
@@ -209,6 +209,34 @@ def validate_telnet_connection(
                 host, port_str, timeout_str, trace_id
             )
 
+    def perform_dns_lookup(
+        self,
+        host: Optional[str],
+        port_str: Optional[str],
+        trace_id: Optional[str] = None,
+    ) -> AgentResponse:
+        """
+        Performs a DNS lookup for the given host name.
+        :param host: Host to check, will raise `BadRequestError` if None.
+        :param port_str: Optional port to pass to `getaddrinfo` API, both int and
+            string are supported.
```

Comment:
would it be more accurate for port to be `Union[int, str]`?

Reply:
good catch, updating it